### ISSUE TITLE
Plugin should only render reStructuredText documents (not any type)

### ISSUE THREAD

**Issue by mcepl** [bittner/gedit-reST-plugin#11]
When I enable the plugin, it tries to work on ALL my files. Which is kind of unfortunate, because it tries to run reST processing even on my Python code etc., which obviously doesn't work well. I have suddenly tons of warning messages like:

```
<string>:75: (ERROR/3) Unexpected indentation.
<string>:79: (WARNING/2) Definition list ends without a blank line; unexpected unindent.
<string>:34: (ERROR/3) Unexpected indentation.
<string>:36: (WARNING/2) Definition list ends without a blank line; unexpected unindent.
<string>:37: (WARNING/2) Block quote ends without a blank line; unexpected unindent.
```

I know that `gedit` plugins are not the same as for example Emacs’ modes, so they natively are not limited just to some particular MIME type, but I guess with the heavy processes run by the plugin (e.g., whole WebKit and OpenJDK running) we should try to emulate this somehow with the infrastrcture we have.

If we moved most of the effective work to the `Gedit.ViewActivatable` child and cover whole processing with something like `if filetype == 'reST':` and ignore every other document. Of course, we would have to add also some event handler for case the user switches the file type of the edited document to reST (I guess we are talking about the file type used for the highlighting in gedit, MIME type is always just `text/plain`, right?).

**Comment by bittner** [comment]
Strange that I never noticed that. Looks like I've always tested with `.rst` files only!  :anguished: 

I don't like the idea of an `if filetype == 'reST':`, there should be a way to have the plugin being hit only when the reST markup is activated. Though, markup and rendering isn't necessarily the same thing, hmmm...

Whatever the solution may look like, file a PR and we'll discuss it.

## Rendering Works For Buffers, Not Only Files

Note that [rendering of reStructuredText](https://github.com/bittner/gedit-reST-plugin/blob/gedit-3.14/reST/restructuredtext.py#L63) is not done only for saved files in this plugin. You can open a blank text view/window and type reStructuredText syntax, and the plugin will generate HTML and display it in the bottom panel.

You can also select only a part of your text document (e.g. a code snippet in an article about reST), and have it rendered in the reStructuredText viewer panel (only the selection is fed to docutils/Sphinx for converting it to HTML).

**Comment by mcepl** [comment]
I don’t like `if filetype == 'reST':` either (I would a way prefer clean Emacs/vim style of filetype modes), but I don’t see how to do it cleanly with gedit.

**Comment by bittner** [comment]
I'm trying to implement the `if filetype == 'reST'` feature now.

Anyone having any idea of [where to get](https://github.com/bittner/gedit-reST-plugin/blob/master/reST/restructuredtext.py#L64) the filetype from?

**Comment by bittner** [comment]
Fixed via d89fdee.

**Comment by mcepl** [comment]
Works fine. Thank you (even with python2 on RHEL-7 with gedit-3.22.0-3).
